### PR TITLE
Add image normalization docs

### PR DESCRIPTION
**Pull request by mattcai**
Added how-tos and tutorials to example gallery in addition to minor edits related to normalization methods. Currently examples are not built by readthedocs because it requires more resources than available on rtd.

### CODE REVIEW

**Comment by mattcai** [comment]
To view the branch on readthedocs: https://spacetx-starfish.readthedocs.io/en/mcai-normalizing-docs/creating_an_image_processing_pipeline/index.html#normalizing-intensities

File: examples/normalize_intensity_distributions.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+.. _tutorial_normalizing_intensity_distributions:
+
+Normalizing Intensity Distributions
+===================================
+
+It is important to normalize images before comparing intensity values between channels and
+rounds to decode feature traces. This tutorial will teach you when and how to normalize the
+*distributions* of intensities from images in an :py:class:`ImageStack`. For more
+background on normalizing images in starfish pipelines see
+:ref:'section_normalizing_intensities`.
```
Comment (neuromusic) [inline examples/normalize_intensity_distributions.py]: this ref is broken

File: examples/normalize_intensity_distributions.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+# Run MatchHistograms
+scaled_cr = mh_cr.run(stack, in_place=False, verbose=False, n_processes=8)
+scaled_c = mh_c.run(stack, in_place=False, verbose=False, n_processes=8)
+scaled_r = mh_r.run(stack, in_place=False, verbose=False, n_processes=8)
+
+####################################################################################################
+# Plotting the intensity histograms shows the effect of normalization.
+#
+# * unscaled histograms of the three channels reflect the raw images -- channel 0 has less signal
+# * normalizing with group_by={Axes.CH} has the effect of significantly rescaling  histograms of channel 0 to match histograms of the other channels
```
Comment (neuromusic) [inline examples/normalize_intensity_distributions.py]: inline code markup here? e.g. `group_by={Axes.CH}`

File: examples/how_to/intensity_histogram.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+important for accurately representing the distribution as a histogram and it may be worth trying
+a couple different bin sizes.
+
+The histogram is useful for examining image data and deciding how to normalize before decoding.
+
+* :ref:`Normalizing Intensity Distributions <tutorial_normalizing_intensity_distributions>`
+* :ref:`Normalizing Intensity Values <tutorial_normalizing_intensity_values>`
+"""
+
+# Load :py:class:`ImageStack` from example DARTFISH data
```
Comment (neuromusic) [inline examples/how_to/intensity_histogram.py]: this line doesn't get rendered by sphinx, so can drop the markup

File: examples/how_to/clip_percentile_to_zero.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+.. _howto_clip_percentile_to_zero:
+
+Clipping Percentile To Zero
+===========================
+
+How to use :py:class:`starfish.image.Filter.ClipPercentileToZero` to clip high and low intensity
+values of image planes or image volumes in an :py:class:`ImageStack`.
+
+:py:class:`ClipPercentileToZero` is the recommended :py:class:`FilterAlgorithm` for clipping images
+for most users. An interval is defined by pixel values at the p_min and p_max percentiles of
```
Comment (neuromusic) [inline examples/how_to/clip_percentile_to_zero.py]: lets wrap kwargs in backticks so they render as code

File: examples/normalize_intensity_distributions.py (hunk 1 above)
Comment (neuromusic) [reply examples/normalize_intensity_distributions.py]: or rather, is not rendering

File: docs/source/creating_an_image_processing_pipeline/index.rst [hunk 5]
```diff
@@ -57,6 +73,88 @@ These approaches ameliorate these problems.
...
+intensity values in one channel might be as high as the signal intensity values of another
+channel. Normalizing image intensities corrects for these differences and allows comparisons to be
+made.
+
+Whether to normalize
+^^^^^^^^^^^^^^^^^^^^
+
+The decision of whether to normalize depends on your data and decoding method used in the next
+step of the pipeline.
+If your :py:class:`ImageStack` has approximately the same range of intensities across rounds and
```
[the remark this answers is not in the document]
Comment (mattcai) [reply docs/source/creating_an_image_processing_pipeline/index.rst]: That works but found better solution by doing `:py:class:`.ImageStack``

File: examples/normalize_intensity_distributions.py (hunk 2 above)
Comment (mattcai) [reply examples/normalize_intensity_distributions.py]: done

File: docs/source/creating_an_image_processing_pipeline/index.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (neuromusic) [reply docs/source/creating_an_image_processing_pipeline/index.rst]: nice!

**Pull request merged**